jwst/resample/resample_spec.py

```diff
@@ -120,13 +120,26 @@ def build_nirspec_output_wcs(self, refmodel=None):
         if refmodel:
             all_wcs.insert(0, refmodel.meta.wcs)
         else:
-            refmodel = self.input_models[0]
+            # Use the first model with any good data as the reference model
+            for model in self.input_models:
+                dq_mask = resample_utils.build_mask(model.dq, self.good_bits)
+                good = np.isfinite(model.data) & (model.data != 0) & dq_mask
+                if np.any(good) and refmodel is None:
```

Comment:
Just curious as to what constitutes a "good" model? Does it take only 1 good (non-zero, non-NaN) pixel within the image?

Reply:
In this case, yes, even one good pixel would be picked up as a "good" reference model.  I wasn't sure what limit might be useful here, so I didn't set one beyond the minimum necessary to make the weighted average over the wavelength and spatial coordinates make sense.